### PR TITLE
Feat/interaction table map

### PR DESCRIPTION
**Pull request by fmarahre**
hover-interaction **table<-->map** works in `CropsView.vue` and `Crops3dView.vue`

### CODE REVIEW

File: vue/src/views/Crops3dView.vue [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
+    </div>
+    <div v-if="storeOptions.indexOf('3d-map') > -1" class="col-8 pl-2">
+      <crops-map
+        ref="cropsMapRef"
+        :crops="crops"
+        @polygon-enter="mapCropsEnter"
+        @polygon-leave="mapCropsLeave"
+      ></crops-map>
     </div>
   </div>
```
Comment (dobro929) [inline vue/src/views/Crops3dView.vue:26]: added view wrapper
```suggestion
  <div class="ml-28">
    <header-bar
      title="3D"
      :widgetOptions="widgetOptions3D"
      :storeOptions="storeOptions"
    ></header-bar>
    <div class="row p-6">
      <div v-if="storeOptions.indexOf('3d-table') > -1" class="col-4">
        <crops-table
          ref="cropsTableRef"
          title="Plants"
          :visibleColumns="columns"
          :crops="crops"
          @row-enter="tableCropsEnter"
          @row-leave="tableCropsLeave"
        ></crops-table>
      </div>
      <div v-if="storeOptions.indexOf('3d-map') > -1" class="col-8 pl-2">
        <crops-map
          ref="cropsMapRef"
          :crops="crops"
          @polygon-enter="mapCropsEnter"
          @polygon-leave="mapCropsLeave"
        ></crops-map>
      </div>
    </div>
  </div>
```

File: vue/src/views/CropsView.vue [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+    </div>
+    <div v-if="storeOptions.indexOf('crops-map') > -1" class="col-4 pl-2">
+      <crops-map
+        ref="cropsMapRef"
+        :crops="crops"
+        @polygon-enter="mapCropsEnter"
+        @polygon-leave="mapCropsLeave"
+      ></crops-map>
     </div>
   </div>
```
Comment (dobro929) [inline vue/src/views/CropsView.vue:26]: added view wrapper
```suggestion
  <div class="ml-28">
    <header-bar
      title="Crops"
      :widgetOptions="widgetOptionsCrops"
      :storeOptions="storeOptions"
    ></header-bar>
    <div class="row p-6">
      <div v-if="storeOptions.indexOf('crops-table') > -1" class="col-8">
        <crops-table
          ref="cropsTableRef"
          title="Overview"
          :crops="crops"
          :visibleColumns="columns"
          @row-enter="tableCropsEnter"
          @row-leave="tableCropsLeave"
        ></crops-table>
      </div>
      <div v-if="storeOptions.indexOf('crops-map') > -1" class="col-4 pl-2">
        <crops-map
          ref="cropsMapRef"
          :crops="crops"
          @polygon-enter="mapCropsEnter"
          @polygon-leave="mapCropsLeave"
        ></crops-map>
      </div>
    </div>
  </div>
```

File: vue/src/views/CropsView.vue [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
 </template>
 
 <script setup lang="ts">
-import { onMounted, type Ref } from "vue";
+import { onMounted, type Ref, ref } from "vue";
```
Comment (dobro929) [inline vue/src/views/CropsView.vue:30]: removed unused import
```suggestion
import { type Ref, ref } from "vue";
```

File: vue/src/views/Crops3dView.vue [hunk 4]
```diff
@@ -1,23 +1,33 @@
...
 </template>
 
 <script setup lang="ts">
-import { onMounted, type Ref } from "vue";
+import { onMounted, type Ref, ref } from "vue";
```
Comment (dobro929) [inline vue/src/views/Crops3dView.vue:30]: removed unused import
```suggestion
import { type Ref, ref } from "vue";
```

File: vue/src/components/CropsTable.vue [hunk 5]
```diff
@@ -223,36 +220,78 @@ import type { Crop } from "../types/crop";
...
+  emit("rowLeave", cropsId);
 }
 
 function rowclicked(): void {
   console.log("rowClicked");
 }
+
+let visCols = ref(props.visibleColumns);
+
+function pagination() {
```
Comment (lucaslootalot) [inline vue/src/components/CropsTable.vue:288]: add return type

**Review by lucaslootalot — CHANGES_REQUESTED** [review]

File: vue/src/components/CropsTable.vue (hunk 5 above)
Comment (fmarahre) [reply vue/src/components/CropsTable.vue]: done

**Review by lucaslootalot — APPROVED** [review]
lgtm, i will squash the commits and then you can merge the PR 👍🏻

**Pull request merged**